[iOS] screen.colorDepth reports incorrect value

annevk opened the pull request:
#### c073b6cbaccdcb73c1b5a70f95e87c9f02043681
<pre>
[iOS] screen.colorDepth reports incorrect value
<a href="https://bugs.webkit.org/show_bug.cgi?id=244845">https://bugs.webkit.org/show_bug.cgi?id=244845</a>
rdar://99871925

Reviewed by Jer Noble.

24 is the CSSOM View value for 8 bits per color channel. The alpha channel is to be excluded.

* LayoutTests/platform/ios-device/TestExpectations:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/css/cssom-view/Screen-pixelDepth-Screen-colorDepth001-expected.txt: Removed.
* Source/WebCore/platform/ios/PlatformScreenIOS.mm:
(WebCore::screenDepth):
(WebCore::screenDepthPerComponent):
(WebCore::screenHasInvertedColors):
(WebCore::screenPPIFactor):
(WebCore::collectScreenProperties):

Canonical link: <a href="https://commits.webkit.org/262386@main">https://commits.webkit.org/262386@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0b02eec292a7152dff8a073592f9f208ae2d06d9

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/956 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/983 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/1020 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/1428 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/849 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/945 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/1160 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/1068 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/1060 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/965 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/1008 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/899 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/1318 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/1049 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/896 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/895 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/873 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/923 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/1939 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/914 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/867 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/861 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/905 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/348 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/928 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

karlcow reviewed (commented):
I can see, how there is a chance to break things, if people have specific code path related to the detection of `screen.colorDepth`.

After searching a bit I don't see any obvious tools using it. We could probably quirk if we meet some breakages here and there. I can imagine fingerprinting libraries could use this. 

So I'm saying OK. Just let's be ready if certain things break. 
Thanks @annevk for the heads up.

annevk committed:
[iOS] screen.colorDepth reports incorrect value
https://bugs.webkit.org/show_bug.cgi?id=244845
rdar://99871925

Reviewed by Jer Noble.

24 is the CSSOM View value for 8 bits per color channel. The alpha channel is to be excluded.

* LayoutTests/platform/ios-device/TestExpectations:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/css/cssom-view/Screen-pixelDepth-Screen-colorDepth001-expected.txt: Removed.
* Source/WebCore/platform/ios/PlatformScreenIOS.mm:
(WebCore::screenDepth):
(WebCore::screenDepthPerComponent):
(WebCore::screenHasInvertedColors):
(WebCore::screenPPIFactor):
(WebCore::collectScreenProperties):

Canonical link: https://commits.webkit.org/262386@main
